llvm/cmake/config-ix.cmake

```diff
@@ -3,6 +3,44 @@ if( WIN32 AND NOT CYGWIN )
   set(PURE_WINDOWS 1)
 endif()
 
+if (CMAKE_SYSTEM_NAME STREQUAL "AIX")
+  set(AIX 1)
```

Comment:
I will humbly suggest to not define any of these variables in global scope and inline the checks.
They look like they can conflict with CMake-predefined variables and / or variables defined in other places.
(note that this file is `include`d, which means the names are not local to this file).

Reply:
Looking at the `PURE_WINDOWS` above, `confix-ix.cmake` seems the right place to provide such variables. If we define them anywhere else outside of code supporting standalone builds, we should stop doing that.

If you look at the latest CMake documentation, you might think we're conflicting here, but none of them are provided by CMake 3.20.

Reply:
> Looking at the `PURE_WINDOWS` above, `confix-ix.cmake` seems the right place to provide such variables.

My point is that we shouldn't provide them and instead check CMAKE_SYSTEM_NAME directly, as I believe is done in most places.

https://github.com/llvm/llvm-project/blob/a9604cd3212beb6cdb41a79f248723140ab9939d/llvm/cmake/modules/AddLLVM.cmake#L336).

These variables only save a couple of lines in the first `if` below.

Reply:
Would you like to remove `PURE_WINDOWS`, too?